engines/sherlock/objects.h

```diff
@@ -0,0 +1,271 @@
+};
+
+class Object {
+private:
+	static SherlockEngine *_vm;
+
+	bool checkEndOfSequence();
+
+	void setObjSequence(int seq, bool wait);
+public:
+	static bool _countCAnimFrames;
+
+	static void setVm(SherlockEngine *vm);
+public:
+	Common::String _name;			// Name
+	Common::String _description;	// Description lines
+	Common::String _examine;		// Examine in-depth description
+	int _sequenceOffset;
+	uint8 *_sequences;				// Holds animation sequences
+	ImageFile *_images;				// Sprite images
+	ImageFrame *_imageFrame;		// Pointer to shape in the images
+	int _walkCount;					// Character walk counter
+	int _allow;						// Allowed menu commands - ObjectAllow
+	int _frameNumber;				// Frame number in rame sequence to draw
+	int _sequenceNumber;			// Sequence being used
+	SpriteType _type;				// Object type
+	Common::Point _position;		// Current position
+	Common::Point _delta;			// Momvement amount
+	Common::Point _oldPosition;		// Old position
+	Common::Point _oldSize;			// Image's old size
+	Common::Point _goto;			// Walk destination
+
+	int _pickup;
+	int _defaultCommand;			// Default right-click command
+	int _lookFlag;					// Which flag LOOK   will set (if any)
+	int _pickupFlag;				// Which flag PICKUP will set (if any)
+	int _requiredFlag;				// Object will be hidden if not set
+	Common::Point _noShapeSize;		// Size of a NO_SHAPE
+	int _status;					// Status (open/closed, moved/not)
+	int8 _misc;						// Misc field -- use varies with type
+	int _maxFrames;					// Number of frames
+	int _flags;						// Tells if object can be walked behind
+	ActionType _aOpen;				// Holds data for moving object
+	AType _aType;					// Tells if this is an object, person, talk, etc.
+	int _lookFrames;				// How many frames to play of the look anim before pausing
+	int _seqCounter;				// How many times this sequence has been executed
+	Common::Point _lookPosition;	// Where to walk when examining object
+	int _lookFacing;				// Direction to face when examining object
+	int _lookcAnim;
+	ActionType _aClose;
+	int _seqStack;					// Allows gosubs to return to calling frame
+	int _seqTo;						// Allows 1-5, 8-3 type sequences encoded in 2 bytes
+	uint _descOffset;					// Tells where description starts in DescText
+	int _seqCounter2;				// Counter of calling frame sequence
+	uint _seqSize;					// Tells where description starts
+	ActionType _aMove;
+	UseType _use[4];
+
+	Object();
+
+	void load(Common::SeekableReadStream &s);
+
+	void toggleHidden();
+
+	void checkObject();
+
+	int checkNameForCodes(const Common::String &name, const char *const messages[]);
+
+	void setFlagsAndToggles();
+
+	void adjustObject();
+
+	int pickUpObject(const char *const messages[]);
+
+	int frameWidth() const { return _imageFrame ? _imageFrame->_frame.w : 0; }
+	int frameHeight() const { return _imageFrame ? _imageFrame->_frame.h : 0; }
+	const Common::Rect getNewBounds() const;
+	const Common::Rect getNoShapeBounds() const;
+	const Common::Rect getOldBounds() const;
+};
+
+struct CAnim {
```

Comment:
Is there any specific reason to call it 'CAnim'? That looks so much like a prefix for being a 'class' or something.